app/main/views/frameworks.py

```diff
@@ -329,31 +330,28 @@ def framework_supplier_declaration(framework_slug, section_id=None):
                     current_user.email_address
                 )
 
-                next_section = content.get_next_editable_section_id(section_id)
-                if next_section:
-                    return redirect(
-                        url_for('.framework_supplier_declaration',
-                                framework_slug=framework['slug'],
-                                section_id=next_section))
+                if next_section and not request.form.get('save_and_return_to_overview', False):
+                    # Go to the next section.
+                    return redirect(url_for(
+                        '.framework_supplier_declaration',
+                        framework_slug=framework['slug'],
+                        section_id=next_section.id
+                    ))
                 else:
-                    url = "{}/declaration_complete".format(
-                        url_for('.framework_dashboard',
-                                framework_slug=framework['slug']))
-                    flash(url, 'declaration_complete')
-                    return redirect(
-                        url_for('.framework_dashboard',
-                                framework_slug=framework['slug']))
+                    # Otherwise they have reached the last page of their declaration.
+                    # Return to the overview.
+                    return redirect(url_for('.framework_supplier_declaration_overview', framework_slug=framework_slug))
             except APIError as e:
                 abort(e.status_code)
 
     return render_template(
         "frameworks/edit_declaration_section.html",
         framework=framework,
+        next_section=next_section,
         section=section,
         declaration_answers=all_answers,
-        is_last_page=is_last_page,
         get_question=content.get_question,
-        errors=errors,
+        errors=errors
```

Comment:
```
x= 'foo', 
len(x)
```

I'm not advocating using trailing commas in places that sequences aren't being specified. And this would be an argument against commas at the ends of lines at all, which we obviously can't do without. There's nothing "trailing"-comma specific about it.

```
('foo')
('foo',)
```

This is exactly a case I would use to argue in *favour* of trailing commas. Firstly, if you want to make a single-element tuple you *have* to use the second form, no way around it, the trailing comma is needed. But then, if you add a second member, the trailing comma is not needed. Remove trailing comma? Leave it? Surely the most consistent thing is to just be able to append another member to the line.

Otherwise you end up with pieces of code like

```
a = ("one",)
b = ("two", "one")
```

which is jarring, even though `b` is just a prepend-ation (?) of `a`.

```
c = ("three", "four")
```

Decide that `c` can lose its first member, selecting and deleting the `"three",` and all of a sudden, bam:

```
c = ("four")
```

You've not got a tuple anymore.

```
x=[foo, bar, baz,]
```

Yes, that's just a quirk of the language - I probably *would* add trailing commas here if the language allowed me (it doesn't).